Packages/com.unity.inputsystem/InputSystem/InputManagerStateMonitors.cs

```diff
@@ -261,8 +261,15 @@ public void SortMonitorsByIndex()
                 // Insertion sort.
                 for (var i = 1; i < signalled.length; ++i)
                 {
-                    for (var j = i; j > 0 && listeners[j - 1].monitorIndex < listeners[j].monitorIndex; --j)
+                    for (var j = i; j > 0; --j)
                     {
+                        // Sort by complexities only to keep the sort stable
+                        // i.e. don't reverse the order of controls which have the same complexity
+                        var firstComplexity = (int)((listeners[j - 1].monitorIndex >> 48) & 0xff);
```

Comment:
I guess its some kind of priority combined with index?

Reply:
I found that monitorIndex isn't really an index but some sort of bit packed field.
It's in InputActionState::ToCombinedMapAndControlAndBindingIndex().

Another alternative here would be to make InputActionState::SplitUpMapAndControlAndBindingIndex  public instead of private. And then extend it to pull out the complexity too. Then the code above would be simpler to read. The downside would be that there would be a bunch of extra unpacking and temporary variables needed here.  Probably this is packed for performance reasons so I wanted the simplest code.